### ISSUE TITLE
Document storage

### ISSUE THREAD

**Issue by j-harbin** [dfo-mar-odis/ScienceSPA#11]

**Comment by j-harbin** [comment]
Deciding where to store documents and manage information related to the backlog items to ensure smooth workflow and easy accessibility.

**Comment by j-harbin** [comment]
So far (for the code at least), we pull information from the following sources:

1. API (Both O&M and Week amount information)
2. Trish's spread sheet (Pay scale based on levels)
3. Cookie information

# 1 is fine, because that's unlikely to change.
We should discuss where to store #2. I've made some updates to this spread sheet to allow for consistency for coding, and it would be a good idea if we were all (myself, Trish, Jeff, etc.) using the same spread sheet. 
# 3 should likely be stored in the private folder you mentioned @gomezcatalina , but maybe we'll confirm today.

In addition to the above three sources, we could also discuss where to store things such as our Scrum Powerpoint, your excel sheet indicating different classifications, code, the PowerBI dashboard etc.

**Comment by gomezcatalina** [comment]
Thanks, Jamie! 

Is it possible read data from data SharePoint folder t solve #2? https://086gc.sharepoint.com/sites/MaritimesSpatialPlanning-ScienceSPA/Shared%20Documents/Forms/AllItems.aspx?FolderCTID=0x012000A473D1428CE09049B7F4C7F64C55A26F&id=%2Fsites%2FMaritimesSpatialPlanning%2DScienceSPA%2FShared%20Documents%2FScience%20SPA%2FData&viewid=c1ded5e0%2Dbd05%2D46fa%2Da618%2Dc410291739b8

**Comment by gomezcatalina** [comment]
Options to discuss:

- repo/code: https://github.com/dfo-mar-odis/ScienceSPA
- Spreadsheets (if they can be read from R) & presentations: sharepoint folder [https://086gc.sharepoint.com/sites/MaritimesSpatialPlanning-ScienceSPA/Shared Documents/Science SPA](https://086gc.sharepoint.com/sites/MaritimesSpatialPlanning-ScienceSPA/Shared%20Documents/Forms/AllItems.aspx?RootFolder=%2Fsites%2FMaritimesSpatialPlanning%2DScienceSPA%2FShared%20Documents%2FScience%20SPA&FolderCTID=0x012000A473D1428CE09049B7F4C7F64C55A26F)
- Cookies: IN Folder (very secure)
- PowerBI dashboards: https://sites.google.com/view/science-collaboration-hubs/dashboards

**Comment by j-harbin** [comment]
Yes! That is possible :).

**Comment by gomezcatalina** [comment]
sweet! Also... How does the API data retrieval process work? Is there an intermediate table or a spreadsheet created to store the data pulled? I am curious if it is possible to have snapshots of all the data pulled from different sources (API, spreadsheets) with timestamps, so that the same intermediate output can be used by different people.

**Comment by j-harbin** [comment]
The data is pulled from the APIs in R and stored as data frames in the code. That being said, it's possible to save those data frames as spread sheets (and created code to automatically update them if the API is updated).

**Comment by gomezcatalina** [comment]
Is it considered good practice to have separate code for recreating data (using APIs and spreadsheets, for instance) and updating plots or PowerBI dashboards? For example, the code that creates the data might only need to be run infrequently, while the code for updating the plots or dashboard may need to be run more frequently during development?

**Comment by j-harbin** [comment]
Yes! Absolutely. It would be good practice to have code scripts that 1) pulls data from API and saves it and 2) Loads the saved code and uses it to create plots, etc. This is not currently how the code is set up, but is a very easy (and I think) high priority fix.
